### PR TITLE
Added docker image for cloudfeeds atomhooper

### PR DESCRIPTION
**Pull request by shub6691**

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+RUN mkdir -p "${CATALINA_HOME}" "${AH_HOME}" "${SAXON_HOME}" /etc/atomhopper/ /var/log/atomhopper/ /etc/cloudfeeds/translation/
+
+WORKDIR ${AH_HOME}
+
+COPY --from=tomcat /usr/local/tomcat ${CATALINA_HOME}
+COPY files/* /etc/atomhopper/
+COPY feedscatalog.xml /etc/feedscatalog/
+COPY ${saxon_lic} ${SAXON_HOME}/saxon-license.lic
+RUN chmod +x ${AH_HOME}/start.sh
```
Comment (deep8103) [inline docker/Dockerfile:29]: Please remove this irrelevant line.

File: docker/files/application-context.xml [hunk 2]
```diff
@@ -0,0 +1,4390 @@
...
+	<map>
+	  <entry key="type" value="eventtype"/>
+	  <entry key="tid"  value="tenantid"/>
+        </map>
+      </property>
+      <property name="delimiter" value=":"/>
+  </bean>
+
+
+  <bean id="concorde-jdbc-new-feed-read-repository-bean" class="org.apache.tomcat.jdbc.pool.DataSource">
```
Comment (deep8103) [inline docker/files/application-context.xml:1215]: Duplicate, replace this instance of _concorde_ with _support_

File: docker/files/atom-server.cfg.xml [hunk 3]
```diff
@@ -0,0 +1,828 @@
...
+         will be incorrect.
+         Scheme should be either "http" or "https"
+     -->
+    <host domain="domain.com" scheme="http" />
+
+    <provider-filters>
+        <provider-filter reference="selective-uri-json-filter"/>
+    </provider-filters>
+
+    <workspace title="support workspace" resource="/support/">
```
Comment (deep8103) [inline docker/files/atom-server.cfg.xml:24]: Duplicate, remove this occurence.

File: docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+FROM adoptopenjdk/openjdk8:alpine-slim
+
+LABEL maintainer="cloudfeeds@rackspace.com" \
+      #Atom Hopper version
+      version="1.9.1" \
+      description="Docker image for Cloudfeeds-Atom Hopper"
+
+#Saxon License file
+ARG saxon_lic
+ENV SAXON_HOME=/etc/saxon \
```
Comment (deep8103) [inline docker/Dockerfile:13]: Failed to compile, change this to single ENV instruction. As given below.

ENV SAXON_HOME=/etc/saxon \
         AH_VERSION=1.9.1 \
         CATALINA_HOME=/opt/tomcat \
         AH_HOME=/opt/atomhopper  \
         PATH=${PATH}:${CATALINA_HOME}/bin:${AH_HOME}

File: docker/Dockerfile (hunk 1 above)
Comment (shub6691) [reply docker/Dockerfile]: Done

File: docker/files/application-context.xml (hunk 2 above)
Comment (shub6691) [reply docker/files/application-context.xml]: done

File: docker/files/atom-server.cfg.xml (hunk 3 above)
Comment (shub6691) [reply docker/files/atom-server.cfg.xml:24]: done

File: docker/Dockerfile (hunk 4 above)
Comment (shub6691) [reply docker/Dockerfile]: done

File: docker/Dockerfile [hunk 5]
```diff
@@ -3,19 +3,19 @@ FROM tomcat:9.0.41-jdk8 as tomcat
...
 ARG saxon_lic
-ENV SAXON_HOME=/etc/saxon \
+ARG app_version
+ARG ah_version
 
-ENV  AH_VERSION=1.9.1 \
-     CATALINA_HOME=/opt/tomcat \
-     AH_HOME=/opt/atomhopper  \
-     PATH=${PATH}:${CATALINA_HOME}/bin:${AH_HOME}
+ENV SAXON_HOME=/etc/saxon
```
Comment (deep8103) [inline docker/Dockerfile:14]: Build failed, compile issues. Replace with:
ENV SAXON_HOME=/etc/saxon \
        AH_VERSION=$ah_version \
        CATALINA_HOME=/opt/tomcat \
        AH_HOME=/opt/atomhopper \
        PATH=${PATH}:${CATALINA_HOME}/bin:${AH_HOME}

File: README.md [hunk 6]
```diff
@@ -1,6 +1,34 @@
 cloudfeeds-atomhopper
 =====================
 
+**docker build image and run the container**
+Your current direcotry should be pointing to cloudfeeds-atomhopper/docker. Run the following command to build an image.
+```
+$docker build -t cloudfeeds-atomhopper:latest-alpine .
```
Comment (deep8103) [inline README.md:7]: This instruction missing mandatory arguments.

File: README.md [hunk 7]
```diff
@@ -1,6 +1,34 @@
...
+```
+$docker run -d --name atomhopper -p 8080:8080 atomhopper:latest-alpine
+```
+These ARGS  should be provided while building the docker image
+```
+ARG saxon_lic
+ARG app_version
+ARG ah_version
+
+For ex: docker build --build-arg ah_version=1.9.1 --build-arg AH_VERSION=1.136.1
```
Comment (deep8103) [inline README.md:20]: misleading instruction. It should have been:
docker build --build-arg ah_version=1.9.1 --build-arg app_version=1.136.1 --build-arg saxon_lic=saxon-license.lic

File: docker/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+COPY files/* /etc/atomhopper/
+COPY feedscatalog.xml /etc/feedscatalog/
+COPY ${saxon_lic} ${SAXON_HOME}/saxon-license.lic
+
+
+RUN apk --no-cache add curl tar bash \
+    && curl https://maven.research.rackspacecloud.com/content/repositories/releases/com/rackspace/usage/usage-schema/${app_version}/usage-schema-${app_version}-schema.tar.gz | tar xz \
+    && mv usage-schema-${APP_VERSION}/xslt-artifacts/* /etc/cloudfeeds/translation/. \
+    && curl -o feeds-atomhopper.war https://maven.research.rackspacecloud.com/content/repositories/releases/com/rackspace/feeds/feeds-atomhopper/${AH_VERSION}/feeds-atomhopper-${AH_VERSION}.war \
+    && mv feeds-atomhopper.war ${CATALINA_HOME}/webapps/ROOT.war \
```
Comment (deep8103) [inline docker/Dockerfile:35]: Remove `\` line number 35. Received warning from docker:
[WARNING]: Empty continuation line found in:
 .
.
.
[WARNING]: Empty continuation lines will become errors in a future release.

**Review by deep8103 — CHANGES_REQUESTED** [review]

File: docker/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+COPY --from=tomcat /usr/local/tomcat ${CATALINA_HOME}
+COPY files/* /etc/atomhopper/
+COPY feedscatalog.xml /etc/feedscatalog/
+COPY ${saxon_lic} ${SAXON_HOME}/saxon-license.lic
+
+
+RUN apk --no-cache add curl tar bash \
+    && curl https://maven.research.rackspacecloud.com/content/repositories/releases/com/rackspace/usage/usage-schema/${app_version}/usage-schema-${app_version}-schema.tar.gz | tar xz \
+    && mv usage-schema-${APP_VERSION}/xslt-artifacts/* /etc/cloudfeeds/translation/. \
```
Comment (deep8103) [inline docker/Dockerfile:33]: Replace AH_VERSION to app_version

File: docker/Dockerfile (hunk 4 above)
Comment (shub6691) [reply docker/Dockerfile]: done.Tested now and build is successful.

File: docker/files/atom-server.cfg.xml (hunk 3 above)
Comment (shub6691) [reply docker/files/atom-server.cfg.xml:24]: done

File: docker/Dockerfile (hunk 5 above)
Comment (shub6691) [reply docker/Dockerfile]: fixed compile issue and build was successful

File: README.md (hunk 6 above)
Comment (shub6691) [reply README.md]: updated .

File: README.md (hunk 7 above)
Comment (shub6691) [reply README.md]: done with updated statement

File: docker/Dockerfile (hunk 8 above)
Comment (shub6691) [reply docker/Dockerfile]: done.

File: docker/Dockerfile (hunk 9 above)
Comment (shub6691) [reply docker/Dockerfile]: done.

**Comment by deep8103** [comment]
Able to build an image by providing proper schema_version that contains xslt artifacts.

**Review by deep8103 — APPROVED** [review]

**Pull request merged**